include/swift/AST/IRGenOptions.h

```diff
@@ -97,8 +102,11 @@ class IRGenOptions {
   /// Which sanitizer is turned on.
   OptionSet<SanitizerKind> Sanitizers;
 
-  /// Whether we should emit debug info.
-  IRGenDebugInfoKind DebugInfoKind : 2;
+  /// Whether we should emit dwarf debug info.
+  IRGenDwarfDebugInfoKind DwarfInfoKind : 2;
+
+  /// Whether we should emit codeview debug info.
+  IRGenCodeViewDebugInfo CodeViewInfo : 1;
```

Comment:
I'm still not happy with this. Can't CodeView do line tables?

Reply:
No, unfortunately, CodeView really has no concept of "levels". You generate full fidelity debug info. The generated information is coalesced into a PDB, which can be post-processed to reduce the amount of information.

Reply:
I think this representation is causing problems elsewhere. We really do want a "level of debug info" setting and a "debug format" setting, not two different "level" settings. (And then probably a separate error saying that CodeView can't do line tables or full type info—the experimental "DWARF types" option.)